servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/CompletableMergeWithPublisher.java

```diff
@@ -67,30 +61,24 @@ void handleSubscribe(final Subscriber<? super T> subscriber, final SignalOffload
         private static final AtomicReferenceFieldUpdater<MergerDelayError, TerminalSignal> terminalUpdater =
                 AtomicReferenceFieldUpdater.newUpdater(MergerDelayError.class, TerminalSignal.class, "terminal");
         private final CompletableSubscriber completableSubscriber;
-        private final Subscriber<? super T> offloadedSubscriber;
+        private final Subscriber<? super T> wrappedSubscriber;
         private final DelayedSubscription subscription = new DelayedSubscription();
         @Nullable
         private volatile TerminalSignal terminal;
 
-        MergerDelayError(Subscriber<? super T> subscriber, SignalOffloader signalOffloader, AsyncContextMap contextMap,
+        MergerDelayError(Subscriber<? super T> subscriber, AsyncContextMap contextMap,
                AsyncContextProvider contextProvider) {
-            // This is used only to deliver signals that originate from the mergeWith Publisher. Since, we need to
-            // preserve the threading semantics of the original Completable, we offload the subscriber so that we do not
-            // invoke it from the mergeWith Publisher Executor thread.
-            this.offloadedSubscriber = signalOffloader.offloadSubscriber(contextProvider.wrapPublisherSubscriber(
-                    subscriber, contextMap));
+            // This is used only to deliver signals that originate from the mergeWith Publisher.
+            this.wrappedSubscriber = contextProvider.wrapPublisherSubscriber(subscriber, contextMap);
             completableSubscriber = new CompletableSubscriber();
         }
 
-        void merge(Completable original, Publisher<? extends T> mergeWith, SignalOffloader signalOffloader,
+        void merge(Completable original, Publisher<? extends T> mergeWith,
                    AsyncContextMap contextMap, AsyncContextProvider contextProvider) {
-            offloadedSubscriber.onSubscribe(
+            wrappedSubscriber.onSubscribe(
                     new MergedCancellableWithSubscription(subscription, completableSubscriber));
-            original.delegateSubscribe(completableSubscriber, signalOffloader, contextMap,
+            original.delegateSubscribe(completableSubscriber, contextMap,
                     contextProvider);
-            // SignalOffloader is associated with the original Completable. Since mergeWith Publisher is provided by
-            // the user, it will have its own Executor, hence we should not pass this signalOffloader to subscribe to
-            // mergeWith.
             // Any signal originating from mergeWith Publisher should be offloaded before they are sent to the
```

Comment:
No, it is correct to remove it. The comment is referring to the use of `subscribeInternal` which would have use the original publisher's SignalOffloader rather than directly calling `handleSubscribe` with what would have been the Completable's SignalOffloader. Interestingly the merged publisher uses a copied AsyncContext rather than sharing AsyncContext which might be debated.